### DISCUSSION TITLE
Proposal: and if

### DISCUSSION THREAD

**Discussion by Astrovolts** [dotnet/csharplang#8602] [Language Ideas]
Good day Gentlemen, 

My suggestion is to implement an `and if` into C#; opposite of the `else if` statement.

Traditional nested if statements:
```
if (car != null)
{
    if (car.HasWheels)
    {
        if (car.HasEngine)
        {
            if (car.IsElectric)
            {
                Console.WriteLine("This is an electric car with wheels and an engine.");
            }
            else
            {
                Console.WriteLine("This is a non-electric car with wheels and an engine.");
            }
        }
        else
        {
            Console.WriteLine("This is a car with wheels but no engine.");
        }
    }
    else
    {
        Console.WriteLine("This is a car with no wheels.");
    }
}
else
{
    Console.WriteLine("No car data provided.");
}
```
This quickly becomes deeply nested and hard to follow as conditions grow.

Using and if to avoid nesting:
```
if (car != null)
{
    Console.WriteLine("Car data provided.");
}
and if (car.HasWheels)
{
    Console.WriteLine("The car has wheels.");
}
and if (car.HasEngine)
{
    Console.WriteLine("The car has an engine.");
}
and if (car.IsElectric)
{
    Console.WriteLine("The car is electric.");
}
else
{
    Console.WriteLine("The car is not electric.");
}
```

**Comment by CyrusNajmabadi** [comment]
C# already has 

```c#
if (dog && brown && tail)
```

Can you use that?

**Comment by TahirAhmadov** [comment]
If you have incrementally more actions depending on each condition:
```cs
if (dog)
{
  Console.WriteLine("It is a dog.");
  if (brown)
  {
    Console.WriteLine("The dog is brown.");
    if (tail)
    {
      Console.WriteLine("The dog has a tail.");
    }
  }
}
```
And if you don't, just merge the nested `if` statements like Cyrus suggested.

**Comment by BlinD-HuNTeR** [comment]
```
if (car == null)
    Console.WriteLine("No car data provided.");
else if (!car.HasWheels)
    Console.WriteLine("This is a car with no wheels.");
else if (!car.HasEngine)
    Console.WriteLine("This is a car with wheels but no engine.");
else if (!car.IsElectric)
    Console.WriteLine("This is a non-electric car with wheels and an engine.");
else
    Console.WriteLine("This is an electric car with wheels and an engine.");
```
P.S.: I don't like Python, but love when I can make blocks without braces and pretend that it's Python.

Also:
```
if (!dog) return;
Console.WriteLine("It is a dog.");

if (!brown) return;
Console.WriteLine("The dog is brown.");

if (!tail) return;
Console.WriteLine("The dog has a tail.");
```

**Reply by TahirAhmadov** [reply]
An electric car with an engine, you say?  

![20241112_141458](https://github.com/user-attachments/assets/04940ee9-9c68-4edf-a2b2-acdb400fa822)
